test/suite/test_rollback_to_stable25.py

```diff
@@ -0,0 +1,302 @@
+        # Search the key to evict it. Use both bookends.
+        v = evict_cursor[1]
+        self.assertEqual(v, self. value_z)
+        v = evict_cursor[2 + my_rle_size]
+        self.assertEqual(v, self. value_z)
+        self.assertEqual(evict_cursor.reset(), 0)
+        s.rollback_transaction()
+        evict_cursor.close()
+
+    def check(self, uri, s, ts, expected):
+        cursor = s.open_cursor(uri)
+        s.begin_transaction('read_timestamp=' + self.timestamp_str(ts))
+        # endpoints should still be in place
+        self.assertEqual(cursor[1], self.value_z)
+        self.assertEqual(cursor[2 + my_rle_size], self.value_z)
+
+        #self.assertEqual(cursor.reset(), 0)
+        #self.prout("at " + str(ts) + ": " + str(expected))
+        #for k, v in cursor:
+        #    self.prout("seen: " + str(k) + ": " + str(v))
+
+        for k in range(2, 2 + my_rle_size):
+            if k in expected:
+                self.assertEqual(cursor[k], expected[k])
+            else:
+                cursor.set_key(k)
+                r = cursor.search()
+                if r != wiredtiger.WT_NOTFOUND:
+                    #self.prout("Oops: " + str(k));
+                    #self.prout("Oops2: " + cursor[k]);
+                    pass
+                self.assertEqual(r, wiredtiger.WT_NOTFOUND)
+        s.rollback_transaction()
+        cursor.close()
+
+    def test_rollback_to_stable25(self):
+        # Create a table without logging.
+        uri = "table:rollback_to_stable25"
+        #format = 'key_format=i,value_format=S'
+        format = 'key_format=r,value_format=S'
+        self.session.create(uri, format + ', log=(enabled=false)')
+
+        # Pin oldest timestamp to 5.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(5))
+
+        # Start stable timestamp at 5.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(5))
+
+        value_a = "aaaaa" * 10
+        value_b = "bbbbb" * 10
+        value_c = "ccccc" * 10
+
+        s = self.conn.open_session()
+
+        # Write the endpoints at time 5.
+        cursor = s.open_cursor(uri)
+        s.begin_transaction()
+        cursor[1] = self.value_z
+        cursor[2 + my_rle_size] = self.value_z
+        s.commit_transaction('commit_timestamp=' + self.timestamp_str(5))
+        self.evict(uri, s)
+        cursor.close()
+
+        # Do writes at time 10.
+        expected10 = self.writes(uri, s, {}, self.type_10, self.write_10, value_a, 10)
+
+        # Evict at time 10 if requested.
+        if self.evict_time == 10:
+            self.evict(uri, s)
+
+        # Do more writes at time 20.
+        expected20 = expected10.copy()
+        expected20 = self.writes(uri, s, expected20, self.type_20, self.write_20, value_b, 20)
+
+        # Evict at time 20 if requested.
+        if self.evict_time == 20:
+            self.evict(uri, s)
+
+        # Do still more writes at time 30.
+        expected30 = expected20.copy()
```

Comment:
I don't understand clearly the need for this copy?

Reply:
self.writes() updates the dict passed to it, because that's how Python works, so because we want to keep a copy of the way things are expected to be at each time, it copies it. However, I can definitely make the code clearer. Let me do that...

Reply:
(done)